Automattic/jetpack
# Add new endpoint `attach-licenses` to Jetpack Plugin endpoints and API `js-package`

#### Changes proposed in this Pull Request:

* Add new method `attach-licenses` to jetpack plugin
* Add support for new `attach-licenses` endpoint to `api` js package
* Add test to `licensing` package to clarify expected return 

#### Jetpack product discussion



Part of the on-going Jetpack License-based checkout project ( p1HpG7-cS-p2 )

#### Does this pull request change what data or activity we track or use?



No

#### Testing instructions:






* Verify new tests pass
* Make requests from console on Jetpack dashboard:

```
jQuery
  .ajax({
    url: "/wp-json/jetpack/v4/licensing/attach-licenses",
    method: "post",
    beforeSend: function (xhr) {
      xhr.setRequestHeader("X-WP-Nonce", Initial_State.WP_API_nonce);
    },
    contentType: "application/json",
    dataType: "json",
    data: JSON.stringify( {
        licenses: [
            'test'
        ]
    })
  })
  .done(function (response) {
    console.log(response);
  })
  .error(function (error) {
    console.log(error.responseText);
  });

```

with an invalid license like above, we would expect an array response with an error inside
<img width="409" alt="Screen Shot 2021-11-09 at 1 38 43 PM" src="https://user-images.githubusercontent.com/2810519/141009151-7de26a39-27ea-4399-ad18-558786194a45.png">

whereas with a proper license we would expect a proper response with the `activatedProductId`:
<img width="444" alt="Screen Shot 2021-11-09 at 1 47 29 PM" src="https://user-images.githubusercontent.com/2810519/141010244-0701056f-e701-418e-b4b5-7e6da12d858b.png">
